README.md

```diff
@@ -100,335 +143,398 @@ immunopepper make_bg \
+
+*Optional parameters to add additional filters*: Parameters to add the additional filters shown in step 1 of both pipelines.
+- `--filterNeojuncCoord`: Default = ''. Choices = ['C', 'N', 'A']. This argument will filter the neojunctions, and it will retain only the kmers that are generated from neojunctions. These are peptides whose junction coordinates were not part of the annotation files. Selecting this option means that only those kmers with junctionAnnotated = False will be considered for further filtering. This filter is used in the preprocessing (step 1 of cancer and normal pipelines). If 'C' is selected, the filter is only applied to the cancer sample. If 'N' is selected, the filter is only applied to the normal samples. If 'A' is selected, the filter is applied to both cancer and normal samples.
+- `--filterAnnotatedRF`: Default = ''. Choices = ['C', 'N', 'A']. This argument will only retrieve kmers that were generated from annotated reading frames and discard those that were obtained by reading frame propagation through the graph. By selecting this option, only kmers generated from reading frames present in the annotation are kept. This means that selecting this option means that only those kmers with ReadFrameAnnotated = True will be selected. This filter is used in the preprocessing (step 1 of cancer and normal pipelines). If 'C' is selected, the filter is only applied to the cancer sample. If 'N' is selected, the filter is only applied to the normal samples. If 'A' is selected, the filter is applied to both cancer and normal samples.
+
+*Optional development parameters*
+- `--tot-batches`: Default = None. If selected, the filtering of the background and foreground will be based on hash functions. This parameter will set the total number of batches in which we will divide the foreground and background files to filter, and each of those batches will be assigned a hash value. If `--batch-id` is specified, `--tot-batches` should also be specified.
+- `--batch-id`: Default = None. If selected, the filtering of the background and foreground will be based on hash functions. This parameter will set the batch id of the current batch that is being filtered. The batch id should be an integer between 0 and `--tot-batches`. It shows the specific batch that we want to process, out of the `--tot-batches`. If `--batch-id` is specified, `--tot-batches` should also be specified.
+- `--on-the-fly`: Default = False. If set to true, all the filtering steps will be done on the fly, without the creation of intermediate files. This will slow down the computations if there is a re-run of the program, as filtering is an expensive operation. However, it will save space in the disk.
```

Comment:
Formulation: This will slow down the computations if there is a re-run of the program
--> The creation of intermediate files would speed up the computations in the case where full or partial reruns are planned. An example of a partial rerun would be when several normal and cancer filtering parameters are applied to the same cohort. Choosing not to save intermediate files will trade speed for disk space.

Reply:
Done